crates/explorer/src/index.js

```diff
@@ -7,155 +7,48 @@ class State {
 
-// Get asm instruction elements by Wasm offset.
-const asmByOffset = new Map();
-
-// Get all (WAT chunk or asm instruction) elements by offset.
-const anyByOffset = new Map();
-
-const addWatElem = (offset, elem) => {
-  if (!watByOffset.has(offset)) {
-    watByOffset.set(offset, []);
-  }
-  watByOffset.get(offset).push(elem);
-
-  if (!anyByOffset.has(offset)) {
-    anyByOffset.set(offset, []);
-  }
-  anyByOffset.get(offset).push(elem);
+const rgbToLuma = (rgb) => {
+  // Use the NTSC color space (https://en.wikipedia.org/wiki/YIQ) to determine
+  // the luminance of this color.  (This is an approximation using powers of two,
+  // to avoid multiplications and divisions. It's good enough for our purposes.)
+  let [r, g, b] = rgbToTriple(rgb);
+  return (((r << 8) + (g << 9) + (b << 7)) >> 10) + (g & 31);
 };
-
-const addAsmElem = (offset, elem) => {
-  if (!asmByOffset.has(offset)) {
-    asmByOffset.set(offset, []);
-  }
-  asmByOffset.get(offset).push(elem);
-
-  if (!anyByOffset.has(offset)) {
-    anyByOffset.set(offset, []);
-  }
-  anyByOffset.get(offset).push(elem);
-};
-
-/*** Event Handlers ************************************************************/
-
-const watElem = document.getElementById("wat");
-watElem.addEventListener("click", event => {
-  if (event.target.dataset.wasmOffset == null) {
-    return;
-  }
-
-  const offset = parseInt(event.target.dataset.wasmOffset);
-  if (!asmByOffset.get(offset)) {
-    return;
-  }
-
-  const firstAsmElem = asmByOffset.get(offset)[0];
-  firstAsmElem.scrollIntoView({
-    behavior: "smooth",
-    block: "center",
-    inline: "nearest",
-  });
-}, { passive: true });
-
-const asmElem = document.getElementById("asm");
-asmElem.addEventListener("click", event => {
-  if (event.target.dataset.wasmOffset == null) {
-    return;
-  }
-
-  const offset = parseInt(event.target.dataset.wasmOffset);
-  if (!watByOffset.get(offset)) {
-    return;
-  }
-
-  const firstWatElem = watByOffset.get(offset)[0];
-  firstWatElem.scrollIntoView({
-    behavior: "smooth",
-    block: "center",
-    inline: "nearest",
-  });
-}, { passive: true });
-
-const onMouseEnter = event => {
-  if (event.target.dataset.wasmOffset == null) {
-    return;
-  }
-
-  const offset = parseInt(event.target.dataset.wasmOffset);
-  const hue = hueForOffset(offset);
-  for (const elem of anyByOffset.get(offset)) {
-    elem.style.backgroundColor = `hsl(${hue} 75% 80%)`;
-  }
+const rgbToTriple = (rgb) => [
+  (rgb >> 16) & 0xff,
+  (rgb >> 8) & 0xff,
+  rgb & 0xff,
+];
+// Get the RGB color for the given offset.  (Memoize to avoid recalculating.)
+const offsetToRgb = new Map();
+const rgbForOffset = (offset) => {
+  let color = offsetToRgb[offset];
+  if (color !== undefined) return color;
+  const crc24 = (crc, byte) => {
+    crc ^= byte << 16;
+    for (let bit = 0; bit < 8; bit++) {
+      crc = ((crc << 1) ^ (crc & 0x800000 ? 0xfa5711 : 0)) & 0xffffff;
+    }
+    return crc;
+  };
```

Comment:
Did you come up with this or did you get it from somewhere else?

Reply:
It's adapted from Wikipedia.  I'll add a note in the code.  (The polynomial is just "fastly" written using hex digits.  It generates surprisingly bright and vibrant colors.)